### PR TITLE
Add Routing example

### PR DESCRIPTION
**Pull request by jontallboy**
Adds an example of using react-router for a cms-react project.

## Screenshots
routing-example-pokedex

### CODE REVIEW

**Review by bmatto — COMMENTED** [review]
Overall this looks great - I think it effectively communicates how to make use of the basepath hook to enable client/server routing.

It would be helpful to have a README that describes the page creation and usage of `[dynamic-slug]`

**Comment by jontallboy** [comment]
> It would be helpful to have a README that describes the page creation and usage of [dynamic-slug]
Ah, yes. Totally forgot about including this like we discussed. I will get that added 👍

File: examples/routing/routing-project/src/components/modules/Router/index.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Island } from '@hubspot/cms-components';
+import AppIsland from '../../islands/App?island';
+
+export const Component = () => {
+  return (
+    <div>
```
Comment (bmatto) [inline examples/routing/routing-project/src/components/modules/Router/index.tsx:7]: Just a curiosity, is this DIV needed here?

Comment (jontallboy) [reply examples/routing/routing-project/src/components/modules/Router/index.tsx:7]: Nope! Just a little leftover cruft 😅

**Comment by jontallboy** [comment]
@bmatto updated to showcase dynamic routes and added a readme calling out the important bits. Let me know what you think!

File: examples/routing/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Routing
+With CMS React, you can use popular routing libraries like React-Router to enable SPA-style routing within your CMS website. The Pokedex example in this directory showcases how to set up SPA routing in your own project using React-Router. This documentation will walk through key aspects of our example, but for more information on React-Router, please visit their [docs](https://reactrouter.com/en/main).
```
Comment (bmatto) [inline examples/routing/README.md:2]: This all looks good, the only thing I think is missing is the usage of [:dynamic-slug] within the actual page to enable the SPA routing on the server.

Comment (jontallboy) [reply examples/routing/README.md:2]: Addressed https://github.com/HubSpot/cms-react/pull/63/commits/299d776986722558d905484b85c7b6a51dc36026

**Pull request merged**